custom_metrics/sass.js

```diff
@@ -0,0 +1,308 @@
+//[sass]
+const SassFunctions = [
+	// Color
+	"color.adjust", "adjust-color",
+	"color.scale", "scale-color",
+	"color.change",
+	"adjust-hue",
+	"opacity",
+	"darken", "lighten",
+	"desaturate", "saturate",
+	"opacify", "transparentize",
+	"fade-in", "fade-out",
+
+	...[
+		"red", "green", "blue",
+		"hue", "saturation", "lightness",
+		"alpha", "complement",
+		"grayscale",
+		"ie-hex-str",
+		"invert",
+		"mix"
+	].flatMap(n => ["color." + n, n]),
+
+	// List
+	"list.separator", "list-separator",
+
+	...[
+		"append", "index", "is-bracketed", "join",
+	 "length", "set-nth", "nth", "zip"
+	].flatMap(n => ["list." + n, n]),
+
+	// Map
+	...[
+		"get", "has-key", "keys", "merge", "remove", "values"
+	].flatMap(n => ["map." + n, "map-" + n]),
+
+	// Math
+	...[
+		"ceil", "floor", "round",
+		"clamp", "min", "max",
+		"abs", "comparable", "unit", "percentage", "random"
+	].flatMap(n => ["math." + n, n]),
+	"math.hypot", "math.log", "math.pow", "math.sqrt",
+	"math.sin", "math.cos", "math.tan",
+	"math.asin", "math.acos", "math.atan", "math.atan2",
+	"math.is-unitless", "unitless",
+
+	// Meta
+	"meta.load-css",
+	"meta.module-functions", "meta.module-variables",
+	...[
+		"call", "get-function", "keywords",
+		"content-exists", "feature-exists", "function-exists", "global-variable-exists", "variable-exists", "mixin-exists",
+		"inspect", "type-of"
+	].flatMap(n => ["meta." + n, n]),
+
+	"if"
+];
+
+const sortObject = obj => Object.fromEntries(Object.entries(obj).sort((a, b) => b[1] - a[1]));
+
+function analyzeSCSS(scss, ret) {
+	if (!scss) {
+		return null;
+	}
+
+	// Drop (most) comments to minimize false positives since we have to parse with regexes
+	// We want to be conservative here and err on the side of NOT removing comments
+	scss = scss.replace(/^\s*\/\*[\s\S]*?\*\//mg, "");
+	scss = scss.replace(/^\s*\/\/.+/mg, "");
+
+	// Most popular variable names
+	ret.variables = {};
+	scss.replace(/\$[\w-]+\b/g, variable => {
+		ret.variables[variable] = (ret.variables[variable] || 0) + 1;
+	});
+	ret.variables = sortObject(ret.variables);
+
+	// Mixins
+	// Note: We are not taking the Indented Mixin Syntax into account
+	ret.mixins = {};
+	scss.replace(/@mixin\s+([\w-]+)(?:\((.+?)\))?/g, ($0, name, args) => {
+		// Note: - and _ are equivalent in mixin names and refer to the same mixin
+		//         we are not taking that into account here. This applies to functions too (below)
+		ret.mixins[name] = {
+			// Default values can't have commas, so split should work
+			args: args?.split(/\s*,\s*/), // TODO separate name from default value?
+			calls: scss.match(RegExp("@include\\s+" + name + "\\b", "gi"))?.length
+		};
+	});
+
+	// Custom functions
+	ret.functions = {};
+	let lastName; // used to track down the nearest @return after a @function
```

Comment:
I'm not sure either, but I thought it might give us insight on what the function does. Good point about multiple returns, I should capture all of them.

Reply:
Alright, it now captures multiple returns. I updated the [gist with the sample data](https://gist.github.com/LeaVerou/5d02efb07945d595b70f5ecdc69057b7) too.
From what I see there is no other `@rule` that takes an `@return`, so it's safe to assume that any `@return` we find after an `@function` is for that function?

Reply:
That sounds right to me, I can't think of a case that would break that rule.